source/python/idetect/scraper.py

```diff
@@ -90,107 +142,81 @@ def format_date(date_string):
     return formatted_date
 
 
-class Scraper(object):
-
-    def __init__(self):
-        pass
+def html_article(url):
+    """Downloads and extracts content plus metadata for html page
+    Parameters
+    ----------
+    url: url of page to be scraped
+
+    Returns
+    -------
+    article: An object of class Article containing the content and metadata.
+    """
+
+    a = newspaper.Article(url)
+    a.download()
+    if a.download_state == 2:
+        a.parse()
+        article_domain = a.source_url
+        article_title = a.title
+        article_authors = a.authors
+        article_pub_date = a.publish_date
+        article_text = remove_newline(a.text)
+        # tag the type of article
+        # currently default to text but should be able to determine img/video
+        # etc
+        article_content_type = 'text'
+        return article_text, article_pub_date, article_title, article_content_type, article_authors, article_domain
+    else:  # Temporary fix to deal with https://github.com/codelucas/newspaper/issues/280
+        return "retrieval_failed", None, "", datetime.datetime.now(), "", ""
+
+
+def get_pdf(url):
+    ''' Takes a pdf url, downloads it and saves it locally.'''
+    try:
+        response = request.urlopen(url)  # not sure if this is needed?
+        publish_date = response.getheader('Last-Modified')
+        pdf_file = open('file_to_convert.pdf', 'wb')
+        pdf_file.write(response.read())
+        pdf_file.close()
+        return os.path.join('./', 'file_to_convert.pdf'), publish_date
+    except (urllib.error.HTTPError, urllib.error.URLError, UnicodeEncodeError, ValueError) as e:
+        return '', ''
+
+
+def get_body_text(url):
+    ''' This function will extract all text from the url passed in
+    '''
+    filepath, publish_date = get_pdf(url)
+    if filepath == '':
+        return '', None
+    else:
+        text = str(textract.process(filepath, method='pdfminer'), 'utf-8')
+        text = text.replace('\n', ' ')  # can replace with a call to
+        text = text.replace('\xa0', ' ')  # the helper function.
+        publish_date = format_date(publish_date)
+        return text, publish_date
+
+
+def remove_pdf(filepath):
+    ''' Deletes pdf from disk
+    Not currently in use as pdfs downloads overwrite self, but may come in 
```

Comment:
I think we should remove code that's not in use. If we need to look back in the git history to see what it was, we can do that.